src/software/new_geom/polygon.h

```diff
@@ -62,7 +62,14 @@ class Polygon : public virtual Shape
     const std::vector<Point>& getPoints() const;
 
    protected:
+    /**
+     * Returns the line segments that connect a list of points.
+     * @return the line segments
+     */
+    static std::vector<Segment> initSegments(std::vector<Point> points);
```

Comment:
Maybe specify the order of points-> segments

Also maybe a @pre of points>1?

Reply:
it should return an empty vector